src/libraries/Microsoft.PowerFx.Core/Binding/BinderCheckTypeResult.cs

```diff
@@ -9,12 +9,34 @@ namespace Microsoft.PowerFx.Core.Binding
 {
     internal readonly struct BinderCheckTypeResult
     {
+        private readonly IReadOnlyCollection<BinderCoercionResult> _coercions;
+
         public bool Success { get; init; }
 
         // The binder has an imperative function called SetCoercedType,
         // which may be called multiple times in a given type checking function
         // This collection represents repeated calls to that function
-        public IReadOnlyCollection<BinderCoercionResult> Coercions { get; init; }
+        public IReadOnlyCollection<BinderCoercionResult> Coercions
+        {
+            get
+            {
+                if (_coercions == null)
```

Comment:
Extremely minor nit, feel free to ignore. 
`return _coercions ?? new List<BinderCoercionResult>();`